Assets/Scripts/PlayerCamera.cs

```diff
@@ -24,6 +41,51 @@ private void Update()
         xRotation = Mathf.Clamp(xRotation, -90f, 90f);
 
         transform.rotation = Quaternion.Euler(0, yRotation, 0);
-        playerCamera.localRotation = Quaternion.Euler(xRotation, 0, 0);
+        playerView.localRotation = Quaternion.Euler(xRotation, 0, 0);
+    }
+
+    private void HeadBobTrigger()
+    {
+        float horizontal = Input.GetAxis("Horizontal");
+        float vertical = Input.GetAxis("Vertical");
+        float inputMagnitude = new Vector3(horizontal, 0, vertical).magnitude;
+
+        if (inputMagnitude > 0)
+        {
+            StartHeadBob();
```

Comment:
This `StartHeadBob` and `StopHeadBob` look weird for me.
just move the logic from `StartHeadBob` to this if
and move the logic from `StopHeadBob` to else